spy_terminal.py

```diff
@@ -27,76 +30,94 @@ def main():
     while x != 'boom':
         if welcome():
             password = set_up()
-            scrambled_message = new_message(codex,count)
+            scrambled_message = new_message(codex, count)
             count = 1
         else:
-            retrieve_message(password, scrambled_message,codex,count)
+            retrieve_message(password, scrambled_message, codex, count)
             scrambled_message = None
             count = 0
         x = input("To self-destruct type 'boom'. Otherwise, hit enter. ")
-    self_destruct(5,'KABOOM', "Yes")
+    self_destruct(5, 'KABOOM', "Yes")
+
 
 '''---------------------SUPPORTING FUNCTIONS ------------------------'''
 
 '''Welcomes the user and directs them to either write a new message or retrieve an existing message.'''
+
+
 def welcome():
     today = date.today()
     print('Welcome Agent. The current date is:', today)
-    desire = input("Enter 'L' to leave a new message. Enter 'R' to retrieve message: ")
+    desire = input(
+        "Enter 'L' to leave a new message. Enter 'R' to retrieve message: ")
```

Comment:
Since this is reused, I'd separate it as its own function. This helps with keeping consistency in user-facing text and makes refactoring when business requirements change easier in the future. It would also allow you to have an isolated unit of code to write an automated test for if you wanted to add unit tests
```suggestion
    desire = get_desire()
```
```py
    def get_desire():
    """
    Prompt user for desired action

    returns string
    """
    return input("Enter 'L' to leave a new message or 'R' to retrieve an existing one: ")
```